From stevenm86@gmail.com Wed May  9 11:53:55 2007
From: Stepan Moskovchenko <stevenm86@gmail.com>
Date: Wed, 09 May 2007 14:53:04 -0400
Subject: USB: ftdi_sio.c: Allow setting latency timer on FT232RL
To: gregkh@suse.de
Cc: linux-usb-devel@lists.sourceforge.net
Message-ID: <1178736784.29161.4.camel@localhost>

From: Stepan Moskovchenko <stevenm86@gmail.com>

The new FT232RL allows setting and getting the value of the latency
timer, like on the FT232BM. However, the driver will not create the
sysfs entries for the RL without this one-line patch. 

I have tested it on two systems with successful results.


From: Stepan Moskovchenko <stevenm86@gmail.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/serial/ftdi_sio.c |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

--- a/drivers/usb/serial/ftdi_sio.c
+++ b/drivers/usb/serial/ftdi_sio.c
@@ -1140,7 +1140,9 @@ static int create_sysfs_attrs(struct usb
 		dbg("sysfs attributes for %s", ftdi_chip_name[priv->chip_type]);
 		retval = device_create_file(&port->dev, &dev_attr_event_char);
 		if ((!retval) &&
-		    (priv->chip_type == FT232BM || priv->chip_type == FT2232C)) {
+		    (priv->chip_type == FT232BM ||
+		     priv->chip_type == FT2232C ||
+		     priv->chip_type == FT232RL)) {
 			retval = device_create_file(&port->dev,
 						    &dev_attr_latency_timer);
 		}
